vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/delete_claim_uphold/delete_claim_uphold.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/endpoint/promotion/delete_claim_uphold/delete_claim_uphold.h"
+
+#include <utility>
+
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/common/request_util.h"
+#include "bat/ledger/internal/endpoint/promotion/promotions_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+
+namespace {
+
+std::string GetPath(const std::string& payment_id) {
+  return base::StringPrintf("/v3/wallet/uphold/%s/claim", payment_id.c_str());
```

Comment:
If the only thing that differs is the path element for the provider, then it might be better to have only one endpoint class and parameterize the `Request` method with the provider (you could just use the wallet type string for now), and then call this file `delete_wallet_claim`.

Reply:
Yep, makes sense.

Reply:
resolved!

Reply:
resolved!